src/seedu/addressbook/data/person/Address.java

```diff
@@ -26,9 +33,65 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
+        
+        String[] addressComponents = trimmedAddress.split(", ");
+       
+        Block block = new Block(Integer.parseInt(addressComponents[BLOCK_INDEX]));
+        Street street = new Street(addressComponents[STREET_INDEX]);
+        Unit unit = new Unit(addressComponents[UNIT_INDEX]);
+        PostalCode postal = new PostalCode(Integer.parseInt(addressComponents[POSTAL_CODE_INDEX]));
+        
+        this.value = block.getBlockNo() + COMMA + street.getStreetName() + COMMA
```

Comment:
Is there a need on keeping this value? Can you reconstruct the value on the fly easily?